src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -246,7 +246,7 @@ boolean shouldSpeculateAndMaybeWait()
         if (cfs.sampleReadLatencyNanos > command.getTimeout(NANOSECONDS))
             return false;
 
-        return !handler.await(cfs.sampleReadLatencyNanos, NANOSECONDS);
+        return true; // do always speculate, be proactive in case a node is replying with a failure STAR-1723
```

Comment:
We should not make this change.  We should not always speculate, that is a waste of resources.